tests/lib/rules/accessor-pairs.js

```diff
@@ -46,38 +324,373 @@ ruleTester.run("accessor-pairs", rule, {
+        },
+        {
+            code: "var o = { get [`a`]() {}, set b(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [setterError, getterError]
+        },
+        {
+            code: "var o = { get [a]() {}, set [b](foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [setterError, getterError]
+        },
+        {
+            code: "var o = { get [a]() {}, set a(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [setterError, getterError]
+        },
+        {
+            code: "var o = { get a() {}, set [a](foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [setterError, getterError]
+        },
+        {
+            code: "var o = { get [a + b]() {}, set [a - b](foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [setterError, getterError]
+        },
+        {
+            code: "var o = { get [`${0} `]() {}, set [`${0}`](foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [setterError, getterError]
+        },
+
+        // Multiple invalid of same and different kinds
+        {
+            code: "var o = { get a() {}, get b() {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [setterError, setterError]
+        },
+        {
+            code: "var o = { set a(foo) {}, set b(bar) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [getterError, getterError]
+        },
+        {
+            code: "var o = { get a() {}, set b(foo) {}, set c(foo) {}, get d() {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [setterError, getterError, getterError, setterError]
+        },
+
+        // Checks per literal
+        {
+            code: "var o1 = { get a() {} }, o2 = { set a(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [setterError, getterError]
+        },
+        {
+            code: "var o1 = { set a(foo) {} }, o2 = { get a() {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [getterError, setterError]
+        },
+
+        // Combinations or valid and invalid
+        {
+            code: "var o = { get a() {}, get b() {}, set b(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [setterError]
```

Comment:
It would be useful if the test could confirm that `a` is the property missing a setter. Could you please add that to the test? (Request applies to similar tests below)

Reply:
Yes, this is definitely an incomplete test that doesn't show much.

This could be tested using location, but it might be also good to change not just the test but the rule itself, to show a message with the name instead of just `Setter is not present.`?

Reply:
I would love to see the property name added to the message, but I won't insist on it happening in this PR.

Reply:
I'll do it, better now in this PR where the tests have to be modified anyway than to fix the whole test file again later.

Just a small question, as there will be different messages for property descriptors (the old generic ones) and object literal properties (which will have the name). Is `messageId` a public data, can I change it?

Reply:
Great question. We haven't really established that `messageId` is in the public API, because we don't yet support message handlers/translators anyway, so I think we can consider the message IDs an implementation detail for now.

Reply:
This is all done now, using `astUtils.getFunctionNameWithKind()`.

The function doesn't return full name for computed ones, just `"getter"` or `"setter"`. This might be a future enhancement, maybe like how [`no-self-assign`](https://github.com/eslint/eslint/blob/master/lib/rules/no-self-assign.js) shows source code without whitespace in the message.

There is a minor bug in `getFunctionNameWithKind()` related to empty string names, I'll fix that in another PR.

Also, the error will now highlight only the header (e.g. `get a`) instead of the whole property, as there is nothing wrong with the code in the getter, it's just missing the setter.

Tests are fixed and should be much better now.